solana/modules/ntt-messages/src/ntt.rs

```diff
@@ -50,22 +51,37 @@ impl Readable for NativeTokenTransfer {
         let to = Readable::read(reader)?;
         let to_chain = Readable::read(reader)?;
 
+        if A::SIZE != Some(0) {
+            // if the size is explicitly zero, this is an empty payload message
+            // and the size field should be skipped
+            // TODO: ditto todo in transceiver.rs
+            let _additional_payload_len: u16 = Readable::read(reader)?;
+        }
+        let additional_payload = A::read_payload(reader)?;
+
         Ok(Self {
             amount,
             source_token,
             to,
             to_chain,
+            additional_payload,
         })
     }
 }
 
-impl Writeable for NativeTokenTransfer {
+impl<A: TypePrefixedPayload + MaybeSpace> Writeable for NativeTokenTransfer<A> {
     fn written_size(&self) -> usize {
         Self::PREFIX.len()
             + TrimmedAmount::SIZE.unwrap()
             + self.source_token.len()
             + self.to.len()
             + ChainId::SIZE.unwrap()
+            + if A::SIZE != Some(0) {
+                u16::SIZE.unwrap()
+            } else {
+                0
+            }
+            + self.additional_payload.written_size()
```

Comment:
Thoughts about having this last line in the `if` block? Not sure if that's more/less explicit

Reply:
I think it makes sense to add it in the `if` block from a code readability perspective.

Reply:
Agree, it also then mirrors the write code.